lightpath/path.py

```diff
@@ -171,86 +176,81 @@ def __init__(self, *devices, name=None):
                             'attribute {}'.format(e))
 
     @property
-    def branches(self):
-        """
-        Branching devices along the path
-        """
-        return [d for d in self.devices if getattr(d, 'branches', False)]
+    def branches(self) -> List[Device]:
```

Comment:
Since we're breaking API and everything here, it may be time for some renames

`branches` -> `branching_devices`?